src/service/resources-impl.js

```diff
@@ -487,6 +487,7 @@ export class Resources {
       resource.requestMeasure();
       dev().fine(TAG_, 'resource reused:', resource.debugid);
     } else {
+      this.relayoutAll_ = true;
```

Comment:
Sorry, now I understand the optimization you mentioned. I don't think it'd work though :( What this bug shows is that there's a DOM order/stubbing randomness. So, it's possible that `resources[i]` actually follows `resources[i + 1]` in both DOM order and layout. So, I'm afraid, we can't make any assumptions of correlation between resource index and DOM/layout placement :(